File: dagger-scope/sample/src/main/java/ru/surfstudio/android/custom_scope_sample/ui/base/configurator/ActivityConfigurator.java
package ru.surfstudio.android.custom_scope_sample.ui.base.configurator;

import ru.surfstudio.android.core.ui.configurator.BaseActivityConfigurator;
import ru.surfstudio.android.custom_scope_sample.app.AppConfigurator;
import ru.surfstudio.android.custom_scope_sample.app.dagger.AppComponent;
import ru.surfstudio.android.custom_scope_sample.ui.base.dagger.activity.ActivityComponent;
import ru.surfstudio.android.custom_scope_sample.ui.base.dagger.activity.ActivityModule;
import ru.surfstudio.android.custom_scope_sample.ui.base.dagger.activity.DaggerActivityComponent;

/**
 * Базовый конфигуратор для активити
 */

public class ActivityConfigurator extends BaseActivityConfigurator<ActivityComponent, AppComponent> {

    @Override
    protected ActivityComponent createActivityComponent(AppComponent parentComponent) {
        return DaggerActivityComponent.builder()
                .appComponent(parentComponent)
                .activityModule(new ActivityModule(getPersistentScope()))
                .build();
    }

    @Override
    protected AppComponent getParentComponent() {
        return AppConfigurator.INSTANCE.getAppComponent();
    }
}

File: template/app-injector/src/main/java/ru/surfstudio/standard/app_injector/ui/configurator/ActivityConfigurator.kt
package ru.surfstudio.standard.app_injector.ui.configurator

import android.content.Context
import ru.surfstudio.android.core.ui.configurator.BaseActivityConfigurator
import ru.surfstudio.standard.app_injector.AppComponent
import ru.surfstudio.standard.app_injector.AppInjector
import ru.surfstudio.standard.app_injector.ui.ActivityComponent
import ru.surfstudio.standard.app_injector.ui.DaggerActivityComponent

class ActivityConfigurator(
        val context: Context
) : BaseActivityConfigurator<ActivityComponent, AppComponent>() {

    override fun createActivityComponent(parentComponent: AppComponent?): ActivityComponent =
            DaggerActivityComponent.builder()
                    .appComponent(parentComponent)
                    .build()

    override fun getParentComponent(): ru.surfstudio.standard.app_injector.AppComponent = AppInjector.appComponent
}